StreamVideoTests/WebRTC/v2/WebRTCAuthenticator_Tests.swift

```diff
@@ -0,0 +1,246 @@
+            .stateAdapter
+            .callSettings
+        XCTAssertTrue(callSettings.audioOn)
+        XCTAssertTrue(callSettings.audioOutputOn)
+    }
+
+    func test_authenticate_updatesVideoOptions() async throws {
+        let create = false
+        let ring = true
+        let expected = JoinCallResponse.dummy(
+            call: .dummy(
+                settings: .dummy(
+                    video: .dummy(
+                        targetResolution: .init(bitrate: 100, height: 200, width: 300)
+                    )
+                )
+            )
+        )
+        mockCoordinatorStack.callAuthenticator.authenticateResult = .success(expected)
+
+        _ = try await subject.authenticate(
+            coordinator: mockCoordinatorStack.coordinator,
+            currentSFU: nil,
+            create: create,
+            ring: ring
+        )
+
+        let videoOptions = await mockCoordinatorStack
+            .coordinator
+            .stateAdapter
+            .videoOptions
+        XCTAssertEqual(videoOptions.preferredDimensions.height, 200)
+        XCTAssertEqual(videoOptions.preferredDimensions.width, 300)
+    }
+
+    func test_authenticate_updatesIntervalOnStatsReporter() async throws {
+        let create = false
+        let ring = true
+        let expected = JoinCallResponse.dummy(
+            statsOptions: .init(reportingIntervalMs: 12000)
+        )
+        mockCoordinatorStack.callAuthenticator.authenticateResult = .success(expected)
+
+        _ = try await subject.authenticate(
+            coordinator: mockCoordinatorStack.coordinator,
+            currentSFU: nil,
+            create: create,
+            ring: ring
+        )
+
+        let statsReporter = await mockCoordinatorStack
+            .coordinator
+            .stateAdapter
+            .statsReporter
+        XCTAssertEqual(statsReporter?.interval, 12)
+    }
+
+    // MARK: - waitForAuthentication
+
+    func test_waitForAuthentication_shouldThrowErrorIfTimeout() async throws {
+        do {
+            try await subject
+                .waitForAuthentication(on: mockCoordinatorStack.sfuStack.adapter)
+            XCTFail()
+        } catch {
+            // Success
+        }
+    }
+
+    func test_waitForAuthentication_shouldWaitUntilConnectionStateIsAuthenticating() async throws {
+        try await withThrowingTaskGroup(of: Void.self) { group in
+
+            group.addTask {
+                try await self.subject
+                    .waitForAuthentication(on: self.mockCoordinatorStack.sfuStack.adapter)
+            }
+
+            group.addTask {
+                await self.wait(for: 0.5)
+                self.mockCoordinatorStack
+                    .sfuStack
+                    .setConnectionState(to: .authenticating)
+            }
+
+            try await group.waitForAll()
+        }
+    }
+
+    // MARK: - waitForConnect
+
+    func test_waitForConnect_shouldThrowErrorIfTimeout() async throws {
+        do {
```

Comment:
wasn't there a way to assert throwing in XCTest? We could skip the do/catch and fail.

Reply:
Good point